get_ending_subscriptions.py

```diff
@@ -0,0 +1,112 @@
+import csv
+from os import environ
+from datetime import datetime, timedelta
+from requests import ConnectionError
+from contentstore.models import MessageSet
+from subscriptions.models import Subscription
+
+from seed_services_client import IdentityStoreApiClient, HubApiClient
+
+
+def get_ending_subs(messageset_name, hub_url, hub_token,
+                    start_date=None, end_date=None,
+                    identity_store_url=environ.get('IDENTITY_STORE_URL'),
+                    identity_store_token=environ.get('IDENTITY_STORE_TOKEN')):
+    #
+    if start_date:
+        start_date = datetime.strptime(start_date, '%Y%m%d')
+    else:
+        start_date = datetime.now()
+    if end_date:
+        end_date = datetime.strptime(end_date, '%Y%m%d') + timedelta(days=1)
+    else:
+        end_date = datetime.now() + timedelta(days=1)
+    #
+    messageset = MessageSet.objects.get(short_name=messageset_name)
+    set_max = messageset.messages.filter(lang="eng_ZA").count()
+    schedule = messageset.default_schedule
+    #
+    # Number of times the schedule will run before the start of the timeframe
+    runs_before_start = len(
+        schedule.get_run_times_between(datetime.now(), start_date))
+    # Number of times the schedule will run before the end of the timeframe
+    runs_before_end = len(
+        schedule.get_run_times_between(datetime.now(), end_date))
+    #
+    # Any schedules older than this will expire before the window
+    max_next_msg = set_max - runs_before_start
+    # Any schedules newer than this will expire after the window
+    min_next_msg = set_max - runs_before_end
+    #
+    subscriptions = Subscription.objects.filter(
+        messageset=messageset, active=True, completed=False,
+        next_sequence_number__gt=min_next_msg,
+        next_sequence_number__lte=max_next_msg)
+    #
+    identity_client = IdentityStoreApiClient(
+        identity_store_token, identity_store_url)
+    hub_client = HubApiClient(hub_token, hub_url)
+    #
+    compile_temp_csv(subscriptions, hub_client, identity_client)
+
+
+def compile_temp_csv(subscriptions, hub_client, identity_client):
+    print("Retrieving and writing data to terminating_subscriptions.csv")
+    with open("terminating_subscriptions.csv", 'w+b') as csvfile:
```

Comment:
We could do either `wb` or `w` here, I can imagine `w+` or `w+b` causing issues here, since if the second time we run this command, if the output is smaller, the file will start with the new output, and then when the new output ends, the old output will follow (since the old output was bigger), and that will cause some weird output files..